### PR TITLE
[MRG] Handle edge case for DAN

### PR DESCRIPTION
**Pull request by YanisLalou**
This pull request includes a change to improve the robustness of the `dan_loss` function. The change include adding a small epsilon to avoid division by zero, and adding a new test case for an edge case.

### Improvements to `dan_loss` function:

* [`skada/deep/losses.py`](diffhunk://#diff-0774706cdd6571bdf61f65dbc2fe0fd5c9e6ae928b28459a523339e962180212L178-R181): Added a small epsilon to the median pairwise distance calculation to avoid potential division by zero issues. (`[skada/deep/losses.pyL178-R181](diffhunk://#diff-0774706cdd6571bdf61f65dbc2fe0fd5c9e6ae928b28459a523339e962180212L178-R181)`)

### Enhancements to testing:

* [`skada/deep/tests/test_deep_divergence.py`](diffhunk://#diff-4a4c6ab899ff86f7bc4d7db095dfc066f1a8b843ac58284dbc9cb8d565b5584bR200-R216): Added a new test case `test_dan_loss_edge_cases` to verify the behavior of `dan_loss` with edge cases, ensuring the function handles zero distances correctly and returns a finite, non-negative loss. (`[skada/deep/tests/test_deep_divergence.pyR200-R216](diffhunk://#diff-4a4c6ab899ff86f7bc4d7db095dfc066f1a8b843ac58284dbc9cb8d565b5584bR200-R216)`)

### CODE REVIEW

File: skada/deep/tests/test_deep_divergence.py [hunk 1]
```diff
@@ -195,3 +197,20 @@ def test_can_with_custom_callbacks():
...
+def test_dan_loss_edge_cases():
+    # Create identical source features to get median distance = 0
+    features_s = torch.tensor([[1.0, 2.0], [1.0, 2.0]], dtype=torch.float32)
+    features_t = torch.tensor([[3.0, 4.0], [5.0, 6.0]], dtype=torch.float32)
+
+    # Verify median distance is 0
+    assert torch.median(torch.cdist(features_s, features_s)) == 0
+
+    # Test that dan_loss still works
+    loss = dan_loss(features_s, features_t)
```
Comment (tgnassou) [inline skada/deep/tests/test_deep_divergence.py:211]: for me the median will not be 0 here no ?

Comment (YanisLalou) [reply skada/deep/tests/test_deep_divergence.py:211]: In `dan_loss` we compute:
`median_pairwise_distance = torch.median(torch.cdist(features_s, features_s))`
And we checked above that's its equal to 0

File: skada/deep/losses.py [hunk 2]
```diff
@@ -175,7 +175,10 @@ def dan_loss(features_s, features_t, sigmas=None):
             In ICML, 2015.
     """
     if sigmas is None:
-        median_pairwise_distance = torch.median(torch.cdist(features_s, features_s))
+        eps = 1e-7
```
[the remark this answers is not in the document]
Comment (YanisLalou) [reply skada/deep/losses.py]: Done

File: skada/deep/tests/test_deep_divergence.py (hunk 1 above)
Comment (tgnassou) [reply skada/deep/tests/test_deep_divergence.py:211]: oh okay! perfect
